packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenuItemWrapper/ContextualMenuAnchor.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import {
+  BaseComponent,
+  anchorProperties,
+  getNativeProps,
+  createRef
+} from '../../../Utilities';
+import { IContextualMenuAnchorProps } from './ContextualMenuAnchor.types';
+import { KeytipData } from '../../KeytipData';
+import { isItemDisabled, hasSubmenu } from '../../../utilities/contextualMenu/index';
+import { IContextualMenuItem, ContextualMenuItem } from '../../ContextualMenu';
+
+export class ContextualMenuAnchor extends BaseComponent<IContextualMenuAnchorProps, {}> {
+  private _anchor = createRef<HTMLAnchorElement>();
+
+  public render() {
+    const {
+      item,
+      classNames,
+      index,
+      focusableElementIndex,
+      totalItemCount,
+      hasCheckmarks,
+      hasIcons,
+      contextualMenuItemAs: ChildrenRenderer = ContextualMenuItem,
+      expandedMenuItemKey,
+      onItemClick,
+      openSubMenu,
+      dismissSubMenu,
+      dismissMenu
+    } = this.props;
+
+    let anchorRel = item.rel;
+    if (item.target && item.target.toLowerCase() === '_blank') {
+      anchorRel = anchorRel ? anchorRel : 'nofollow noopener noreferrer';  // Safe default to prevent tabjacking
+    }
+
+    const subMenuId = this._getSubMenuId(item);
+    const itemHasSubmenu = hasSubmenu(item);
+    const nativeProps = getNativeProps(item, anchorProperties);
+    const disabled = isItemDisabled(item);
+
+    let { keytipProps } = item;
+    if (keytipProps && itemHasSubmenu) {
+      keytipProps = {
+        ...keytipProps,
+        hasMenu: true
+      };
+    }
+
+    return (
+      <div>
+        <KeytipData
+          keytipProps={ item.keytipProps }
+          ariaDescribedBy={ (nativeProps as any)['aria-describedby'] }
+          disabled={ disabled }
+        >
+          { (keytipAttributes: any): JSX.Element => (
+            <a
```

Comment:
ContextualMenuAnchor and ContextualMenuButton are very similar (a different tag is used and a few different props are used). Did you consider creating a generic component that takes a tag and itemprops and would return the correct type of item?

Reply:
This isn't a bad idea, you could even have the tag be a prop that get's passed down.

Reply:
Jeremy and I talked about it; in my opinion there are still a few differences between the two and may require more work. I don't know if this PR should be that far reaching. I understand there's now some duplication across these two files, but I guess I would rather log a bug to fix this later

Reply:
Can you reply with the bug info for the future work?

Reply:
https://github.com/OfficeDev/office-ui-fabric-react/issues/4800